src/commands/dora/__tests__/deployment.test.ts

```diff
@@ -0,0 +1,161 @@
+        },
+      })
+    })
+    test('with minimal parameters provided', async () => {
+      /* eslint-disable prettier/prettier */
+      const {context, code} = await runCLI([
+        '--dry-run', '--skip-git',
+        '--service', 'test-service',
+        '--started-at', '1699960648',
+      ])
+      /* eslint-enable prettier/prettier */
+      expect(code).toBe(0)
+      checkDryRunConsoleOutput(context.stdout, {
+        service: 'test-service',
+        startedAt: new Date(1699960648000),
+        finishedAt: fakeCurrentDate,
+      })
+    })
+    test('with parameters from env', async () => {
+      const envVars = {
+        DD_SERVICE: 'different-test-service',
+        DD_ENV: 'test-env',
+      }
+      /* eslint-disable prettier/prettier */
+      const {context, code} = await runCLI([
+        '--dry-run', '--skip-git',
+        '--started-at', '1699960648',
+      ], envVars)
+      /* eslint-enable prettier/prettier */
+      expect(code).toBe(0)
+      checkDryRunConsoleOutput(context.stdout, {
+        service: envVars.DD_SERVICE,
+        startedAt: new Date(1699960648000),
+        finishedAt: fakeCurrentDate,
+        env: envVars.DD_ENV,
+      })
+    })
+    test('with automatic git info', async () => {
+      const mockGitRepositoryURL = jest.spyOn(gitFunctions, 'gitRepositoryURL')
+      const mockGitHash = jest.spyOn(gitFunctions, 'gitHash')
+      const gitInfo = {
+        repoURL: 'https://github.com/DataDog/datadog-ci',
+        commitSHA: '2186e1b0ab2a87e312a8c831d5bc947fa081d4f9',
+      }
+      mockGitRepositoryURL.mockResolvedValue(gitInfo.repoURL)
+      mockGitHash.mockResolvedValue(gitInfo.commitSHA)
+      /* eslint-disable prettier/prettier */
+      const {context, code} = await runCLI([
+        '--dry-run',
+        '--service', 'test-service',
+        '--started-at', '1699960648',
+      ])
+      /* eslint-enable prettier/prettier */
+      expect(code).toBe(0)
+      expect(mockGitRepositoryURL).toHaveBeenCalled()
+      expect(mockGitHash).toHaveBeenCalled()
+      checkDryRunConsoleOutput(context.stdout, {
+        service: 'test-service',
+        startedAt: new Date(1699960648000),
+        finishedAt: fakeCurrentDate,
+        git: gitInfo,
+      })
+    })
+    test('service is required', async () => {
+      /* eslint-disable prettier/prettier */
+      const {context, code} = await runCLI([
+        '--dry-run', '--skip-git',
+        '--started-at', '1699960648',
+      ])
+      /* eslint-enable prettier/prettier */
+      expect(code).not.toBe(0)
+      expect(context.stdout.toString()).toContain('--service')
+    })
+    test('started-at is required', async () => {
+      /* eslint-disable prettier/prettier */
+      const {context, code} = await runCLI([
+        '--dry-run', '--skip-git',
+        '--service', 'test-service',
+      ])
+      /* eslint-enable prettier/prettier */
+      expect(code).not.toBe(0)
+      expect(context.stdout.toString()).toContain('--started-at')
```

Comment:
is checking this enough? It seems like it could be included even if it does not fail?

Reply:
it's also checking that the exit code is not 0 so I think it's good enough. I don't really have any custom code for this it only prints the usage. It's just because I'm paranoid with clipanion